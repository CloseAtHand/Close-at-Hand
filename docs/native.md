# reactnative 1강

## 프로젝트 시작하기

### 1. native 전역설정
  $ npm install -g expo-cli

### 2. expo 설치 확인
  $ expo
  해당 명령어가 오류 없이 실행되면 완료

### 3. 프로젝트 시작하기
  $ npx create-expo-app demo-project

## 파일 분석

1. .expo-shared 폴더
2. .vscode 폴더