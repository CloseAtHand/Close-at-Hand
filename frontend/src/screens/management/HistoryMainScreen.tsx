import { Pressable, StyleSheet, Text, View } from "react-native";
import { Ionicons } from "@expo/vector-icons";

import { ClothesHistoryList } from "../../components/";
import { FONTSIZE } from "../../shared";
import { useNavigation } from "@react-navigation/native";

const HistoryMainScreen = () => {
	const navigation = useNavigation<Navigation>();

	function handleRecentPress() {
		// 최근 코디 전체보기로 이동
	}

	return (
		<>
			<View style={styles.titleContainer}>
				<Text style={styles.text}>최근 코디</Text>
				<Pressable onPress={handleRecentPress}>
					<Ionicons name="albums-sharp" size={FONTSIZE.Medium} color="black" />
				</Pressable>
			</View>
			<ClothesHistoryList />
		</>
	);
};

export default HistoryMainScreen;

const styles = StyleSheet.create({
	titleContainer: {
		flexDirection: "row",
		alignItems: "center",
		justifyContent: "space-between",
		marginHorizontal: 15,
	},
	text: {
		fontSize: FONTSIZE.ExtarLarge,
		fontWeight: "bold",
	},
});
