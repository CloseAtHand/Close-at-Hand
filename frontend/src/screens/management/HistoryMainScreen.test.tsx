import renderer from "react-test-renderer";
import "@react-navigation/native";

import HistoryMainScreen from "./HistoryMainScreen";

jest.mock("@react-navigation/native");
jest.mock("../../components");

describe("HistoryMainScreen", () => {
	it("스냅샷 테스트", () => {
		const tree = renderer.create(<HistoryMainScreen />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
