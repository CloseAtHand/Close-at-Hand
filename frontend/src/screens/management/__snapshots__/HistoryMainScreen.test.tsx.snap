// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryMainScreen 스냅샷 테스트 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginHorizontal": 15,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 32,
            "fontWeight": "bold",
          }
        }
      >
        최근 코디
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text />
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginHorizontal": 15,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 32,
            "fontWeight": "bold",
          }
        }
      >
        이번 달 가장 많이 입은 옷
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
